lib/active_admin/filters/resource_extension.rb

```diff
@@ -107,7 +107,11 @@ def filter_lookup
 
       # @return [Array] The array of default filters for this resource
       def default_filters
-        default_association_filters + default_content_filters + custom_ransack_filters
+        [].tap do |result|
+          result.concat default_association_filters if namespace.include_default_association_filters
+          result.concat default_content_filters
+          result.concat custom_ransack_filters
+        end
```

Comment:
I was aiming more for cleanliness than beginner friendliness, I apologize. I've altered my code.

I will still stick with `concat` since `<<` produces a different and undesirable result and I think it is preferable to avoid variable names which match method names so I will leave out changing the variable name to `filter`.